compiler/types.nim

```diff
@@ -1632,6 +1635,23 @@ proc isTupleRecursive*(t: PType): bool =
   var cycleDetector = initIntSet()
   isTupleRecursive(t, cycleDetector)
 
+proc isRecursivePointer(typ: PType, isPointer: bool): bool =
```

Comment:
Why the ``isPointer`` here? Isn't that the proc's job to decide?

Reply:
AFAICT this is only used for recursing, the exported overload without the `isPointer` argument below is exported

Reply:
I thought about is. Initially I had a version with ``value is SomePointer and value.isRecursive``. But it didn't work out because of distinct types. A distinct ref not being a ref anymore really hurts here. Neither ``isNil`` nor `` == nil`` work on a distinct ref. My next thought was: "I only really care about cyclic types when the type is also a pointer type, value types can't be cyclic by definition". So I move the "distinct" resolution to the compiler because it was eventually simpler, and probably also a faster implementation.

Reply:
I don't understand this explanation. Rename the ``isPointer`` parameter to something that makes it obvious.

Reply:
This function is recursive. And ``isPointer`` is just the state of the recursive fold on the type tree. After all we don't care about the result of ``canFormAcycle`` isn't a pointer type after all.

Reply:
@Aruq would it help if I rename this proc to ``isPointerAux``?